apps/go-mod-validator/src/go-mod-validator.ts

```diff
@@ -43,13 +43,29 @@ export async function run(): Promise<string> {
 
   const depsToValidate = await getDeps(goModDir, depPrefix);
 
-  const errs: Map<GoModule, string> = new Map();
-
+  const errs: Map<BaseGoModule, string> = new Map();
   const validating = depsToValidate.map(async (d) => {
-    const isValid = await isGoModReferencingDefaultBranch(d, gh);
+    // Bit of a code smell, but I wanted to avoid adding the defaultBranchGetter to deps.ts to keep it separate from
+    // the GitHub API client.
+    // And we want the default branch available in this scope for context.
+    const defaultBranch = await getDefaultBranch(gh, d);
+    const isValid = await isGoModReferencingDefaultBranch(d, defaultBranch, gh);
+
+    let parsedVersion = "UNKNOWN";
+    if ("commitSha" in d) {
+      parsedVersion = d.commitSha;
+    }
+    if ("tag" in d) {
+      parsedVersion = d.tag;
+    }
 
     if (!isValid) {
-      errs.set(d, "dependency not on default branch");
+      errs.set(
+        d,
+        `[${d.goModFilePath}] dependency ${d.name} not on default branch.
+Default branch: ${defaultBranch}
+Version: ${parsedVersion}`,
```

Comment:
Does `d.tag ?? d.commitSha ?? "UNKNOWN"` work here?

Reply:
Unfortunately not, the typechecker is not smart enough to realize that im still handling the unions correctly